### PR TITLE
US 50 & 51

### PR DESCRIPTION
**Pull request by dborski**
## US50: Merchant fulfills part of order
* When each item is fulfilled, it permanently reduces that item's inventory by item_order quantity

## US51: Merchant cannot fulfill an order due to lack of inventory
* A notice now replaces fulfill item button if item does not have enough inventory to cover quantity requested by user

## Other
* Move `item_orders` controller to be under merchant namespace
* Add spec for `full_address` user model method

### CODE REVIEW

File: app/views/merchant/orders/show.html.erb [hunk 1]
```diff
@@ -16,9 +16,12 @@
...
     <p>Quantity: <%= item_order.quantity %></p>
     <p>Item Order Status: <%=item_order.status %></p>
-    <% if item_order.status == "unfulfilled" %>
-      <%= button_to 'Fulfill Item', "/item_orders/#{item_order.id}", method: :patch %>
+    <% if (item_order.status == "unfulfilled") && (item_order.quantity <= item_order.item.inventory) %>
+      <%= button_to 'Fulfill Item', "/merchant/item_orders/#{item_order.id}", method: :patch %>
+    <% elsif item_order.status == "fulfilled" %>  
+      <p>This item has been fulfilled</p>
+    <% else %>
+      <p>Not enough inventory to fulfill item</p>
```
Comment (dborski) [inline app/views/merchant/orders/show.html.erb:24]: I feel this conditional is working how I want, but wouldn't mind someone double checking on heroku.

**Comment by foleymichelle9** [comment]
Looks good! I approve merging!

File: app/controllers/merchant/item_orders_controller.rb [hunk 2]
```diff
@@ -15,6 +16,10 @@ def update_order(order)
       order.update(status: 1)
     end
   end
-  
 
+  def update_item(item_order)
+    item = item_order.item
+    new_inventory = item.inventory - item_order.quantity
```
Comment (perryr16) [inline app/controllers/merchant/item_orders_controller.rb:22]: this looks famliliar

File: app/views/merchant/orders/show.html.erb (hunk 1 above)
Comment (perryr16) [reply app/views/merchant/orders/show.html.erb:24]: I'd recommend a model method for this. Its a bit busy for a view (in my opinion)

Comment (dborski) [reply app/views/merchant/orders/show.html.erb:24]: I'll look into it on a refactor

File: app/controllers/merchant/item_orders_controller.rb (hunk 2 above)
Comment (dborski) [reply app/controllers/merchant/item_orders_controller.rb:22]: Learning from the best

File: app/views/merchant/orders/show.html.erb (hunk 1 above)
Comment (dborski) [reply app/views/merchant/orders/show.html.erb]: @perryr16 I have now updated this with a model method

**Pull request merged**